server/src/test/java/org/elasticsearch/action/ingest/SimulatePipelineResponseTests.java

```diff
@@ -120,4 +92,83 @@ public void testSerialization() throws IOException {
             }
         }
     }
+
+    public static SimulatePipelineResponse createInstance(String pipelineId, boolean isVerbose, boolean withByteArraySource) {
+        int numResults = randomIntBetween(1, 10);
+        List<SimulateDocumentResult> results = new ArrayList<>(numResults);
+        for (int i = 0; i < numResults; i++) {
+            boolean isFailure = randomBoolean();
+            IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), withByteArraySource);
+            if (isVerbose) {
+                int numProcessors = randomIntBetween(1, 10);
+                List<SimulateProcessorResult> processorResults = new ArrayList<>(numProcessors);
+                for (int j = 0; j < numProcessors; j++) {
+                    String processorTag = randomAlphaOfLengthBetween(1, 10);
+                    SimulateProcessorResult processorResult;
+                    if (isFailure) {
+                        processorResult = new SimulateProcessorResult(processorTag, new IllegalArgumentException("test"));
+                    } else {
+                        processorResult = new SimulateProcessorResult(processorTag, ingestDocument);
+                    }
+                    processorResults.add(processorResult);
+                }
+                results.add(new SimulateDocumentVerboseResult(processorResults));
+            } else {
+                results.add(new SimulateDocumentBaseResult(ingestDocument));
+                SimulateDocumentBaseResult simulateDocumentBaseResult;
+                if (isFailure) {
+                    simulateDocumentBaseResult = new SimulateDocumentBaseResult(new IllegalArgumentException("test"));
+                } else {
+                    simulateDocumentBaseResult = new SimulateDocumentBaseResult(ingestDocument);
+                }
+                results.add(simulateDocumentBaseResult);
+            }
+        }
+        return new SimulatePipelineResponse(pipelineId, isVerbose, results);
+    }
+
+    @Override
+    protected SimulatePipelineResponse createTestInstance() {
+        boolean isVerbose = randomBoolean();
+        // since the pipeline id is not serialized with XContent we set it to null for equality tests.
+        return createInstance(null, isVerbose, false);
+    }
+
+    @Override
+    protected SimulatePipelineResponse doParseInstance(XContentParser parser) {
+        return SimulatePipelineResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
I would prefer that we return true here and declare what are the levels that we need to skip when inserting random fields. This is good so that we test that we are forward compatibile where we can be.

Reply:
I will see how this can be done.